### PR TITLE
Boss talk

### PR DESCRIPTION
**Pull request by Aryan-S-Gupta**
# Description

This update includes several changes aimed at improving the testing suite for the NPC components of the project. The key changes are as follows:

- **Added Basic NPCConfigsTest**: Introduced a new test class to validate the configuration of NPCs.
- **Removed MapFactoryTest**: Deleted a test that lacked a class (MapFactory doesnt exist anymore), ensuring clarity and relevance in the test suite.
- **Fixed WeaponFactoryTest**: Addressed issues in the previously commented-out `WeaponFactoryTest` to ensure its functionality (Minor changes).
- **Added Test Case for NPCDeathHandler**: Developed a basic test case for the `NPCDeathHandler` to verify its behavior and correctness.

These changes are motivated by the need to enhance code reliability and ensure comprehensive testing coverage for NPC functionalities.

Fixes / Closes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

The following tests were executed to verify the changes:

- **Basic NPCConfigsTest**: Verified the NPC configurations.
- **NPCDeathHandler Test**: Checked the NPC death handling functionality.

Please follow these steps to reproduce the tests:

1. Run the entire test suite using the command: `./gradlew test`.
2. Ensure that all tests, including the newly added ones, pass without issues.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/test/com/csse3200/game/entities/factories/WeaponFactoryTest.java
Comment (vsinh14) [inline source/core/src/test/com/csse3200/game/entities/factories/WeaponFactoryTest.java:1]: remove the changes its not our tests

**Review by vsinh14 — COMMENTED** [review]
remove weaponfactorytest changes rest of it is good

**Comment by Aryan-S-Gupta** [comment]
Adjusted Accordingly

**Review by vsinh14 — APPROVED** [review]
LGTM

**Pull request merged**